### DISCUSSION TITLE
[운영체제 6,7일차] 프로세스 동기화&데드락

### DISCUSSION THREAD

**Discussion by seongeun42** [Shortudy-10th/tech-for-developer#89] [오늘의 주제]
프로세스 동기화와 데드락 질문 2인조 2개, 3인조 4개

**Comment by LireEruel** [comment]
세마포어와 뮤텍스의 차이점은 무엇인가요?

**Comment by LireEruel** [comment]
데드락의 4가지 필요 조건은 무엇인가요?

**Comment by Jihn0118** [comment]
프로세스 동기화가 무엇이고, 프로세스 동기화를 하는 이유에 대해 설명해주세요.

**Comment by Jihn0118** [comment]
교착 상태가 무엇인가요?

**Reply by seongeun42** [reply]
이거 어떤 상황에서 발생하는지는 위에 있는 데드락 발생 조건 4가지랑 겹치니까 무엇인지까지만 질문으로 할게요~

**Reply by Jihn0118** [reply]
네 수정했습니다

**Comment by seongeun42** [comment]
공유 자원이란 무엇인가요?

**Reply by zini9188** [reply]
여기까지 6일차?

**Comment by seongeun42** [comment]
임계 구역이란 무엇인가요?

**Comment by seongeun42** [comment]
임계 구역 해결 조건에 대해 설명해주세요.

**Comment by seongeun42** [comment]
교착상태 해결 방법의 종류에 대해 말해주세요.

**Comment by aeroej** [comment]
회피 기법인 은행원 알고리즘에 대해 설명해보세요.

**Comment by aeroej** [comment]
Race Condition(경쟁 상태)가 무엇이며, 왜 문제가 되는지 설명해주세요.
